go/vt/dbconnpool/connection_pool.go

```diff
@@ -139,6 +200,16 @@ func (cp *ConnectionPool) Get(ctx context.Context) (*PooledDBConnection, error)
 	if err != nil {
 		return nil, err
 	}
+
+	// Check that the RemoteAddr is still a valid Address
+	if cp.resolutionFrequency > 0 &&
+		net.ParseIP(cp.info.Host) == nil &&
+		!cp.validAddress(net.ParseIP(r.(*PooledDBConnection).RemoteAddr().String())) {
```

Comment:
This could become a problem in the future because every `Get` will end up calling `validAddress` if the feature is enabled. A more sophisticated approach would be to set an epoch on the dns address and reconnect if the connection's epoch doesn't match the current epoch.

Not sure if it's worth it yet.

Reply:
Since I have the DNS resolving in the background this was designed to always check if the address is valid on `Get`. We need some kind of check on every `Get` and this one seemed like a reasonable one as it is a comparison with what should be a short list